src/test/java/javaslang/collection/AbstractMapTest.java

```diff
@@ -99,6 +99,11 @@
         return IntMap.of(emptyMap());
     }
 
+    @Override
+    protected boolean emptyShouldBeSingleton() {
+        return false;
```

Comment:
Yes they are singleton. The tests defined in AbstractTraversableTest don't work for Maps because AbstractMapTest::empty() returns an IntMap. I add another method emptyMapShouldBeSingleton to manage on which subclasses of Map the tests should be run.